pkg/controller/common/helpers.go

```diff
@@ -658,3 +661,120 @@ func dedupePasswdUserSSHKeys(passwdUser ign2types.PasswdUser) ign2types.PasswdUs
 
 	return passwdUser
 }
+
+// CalculateConfigFileDiffs compares the files present in two ignition configurations and returns the list of files
+// that are different between them
+func CalculateConfigFileDiffs(oldIgnConfig, newIgnConfig *ign3types.Config) []string {
+
+	// Go through the files and see what is new or different
+	oldFileSet := make(map[string]ign3types.File)
+	for _, f := range oldIgnConfig.Storage.Files {
```

Comment:
Not sure if it'd be worth the effort or an improvement, but when I was looking at this I was wondering about using 2 loops instead of 4: one loop to construct the `oldFileSet`, and then one loop to simultaneously add every new file and remove every unchanged file from that set